RagtagOpen/freefrom-compensation-api
# Add CMS to our Rails app

This is... a really big PR. Sorry! Here's what it does:
* Converts this previously API-only app to a full Rails app
* Removes the [knock](https://github.com/nsarno/knock) authentication gem (aka remove JWT-based authentication)
* Adds the [devise](https://github.com/plataformatec/devise) authentication gem (add in-app authentication)
* Adds the [administrate](https://github.com/nsarno/knock) CMS gem and customizes dashboards for all of our models.

Here's how to test it:
* Run `bundle exec rake db_migrate`
* Run `bundle exec rake import_data` if you don't already have data imported locally
* Start a rails console: `bundle exec rails c`
  * Create a new user: `User.create(email: 'admin@freefrom.org', password: '12345678')` (or whatever login credentials you want)
* Run the rails app
* Go to localhost:3000 or localhost:3000/admin in the browser
* Log in using your user credentials
* Play around with the new CMS! Load the frontend app as well to see how your changes look in real time :)

Here are some notes on choices I made:
* I decided NOT to create user roles. Instead, I removed the ability for anybody to register users and gave any user the ability to create/edit/delete content. My intention is to create one user for the FreeFrom team myself and then have them share login credentials.
* I removed JWT-based authentication because we currently have no plans to use it (i.e. no plans to add user accounts to the web app)
* I have not deleted the `/data` folder just in case this somehow messes up the data and we need a quick backup.

Closes #68
